theano/gpuarray/c_code/dnn_fwd.c

```diff
@@ -309,7 +333,52 @@ APPLY_SPECIFIC(conv_fwd)(PyGpuArrayObject *input, PyGpuArrayObject *kerns,
       cuda_exit(c->ctx);
       return 1;
     }
+    // save worksize for next time/cache
+    prev_algo.wsSize = worksize;
 
+    // Add to the cache
+    if (params->choose_algo)
```

Comment:
Why, we have successfully used cache hits across many (repeatedly created) modules with same parameters and choose_once.